lua/obsidian/commands/extract_note.lua

```diff
@@ -0,0 +1,40 @@
+local log = require "obsidian.log"
+local util = require "obsidian.util"
+
+---Extract the selected text into a new note
+---and replace the selection with a link to the new note.
+---
+---@param client obsidian.Client
+return function(client, data)
+  local viz = util.get_visual_selection()
+  if viz.lines == nil or #viz.lines == 0 then
+    log.err "ObsidianExtractLink must be called with visual selection"
+    return
+  elseif #viz.lines ~= 1 then
+    log.err "Only in-line visual selections allowed"
+    return
+  end
+
+  local content
+  if string.len(data.args) > 0 then
+    content = { data.args }
+  else
+    content = viz.lines
+  end
+
+  -- new note with title from user
+  local title = vim.fn.input { prompt = "Enter title (optional): " }
+  if string.len(title) == 0 then
+    title = nil
+  end
+  local note = client:new_note(title, nil, client.buf_dir)
+
+  -- replace selection with link to new note
+  local link = client:format_link(note, {})
```

Comment:
`opts` can be nil now
```suggestion
  local link = client:format_link(note)
```

Reply:
True. The type annotations should be updated though.